<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<node pkg="battery_simulator" type="battery" name="battery" output="screen">
		<!-- This node publishes to the topics power (Int32), voltage (Float32), percent (Int32) and recharging (Bool) -->
		<!-- This node subscribes to the topic recharge (Bool) -->
		<!-- Battery discharge model (exponential, linear, inverse), default = exponential -->
		<!-- Reference: https://batteryuniversity.com/learn/article/discharge_methods -->
		<param name="discharge_model" value="exponential" />
		<!-- Max battery voltage, default = 12.5 -->
		<param name="max_voltage" value="12.5" />
		<!-- Min battery voltage, default = 11.5 -->
		<param name="min_voltage" value="11.5" />
		<!-- Base battery voltage, default = 12.0 -->
		<param name="base_voltage" value="12.0" />
		<!-- Initial battery percent charge, default = 100 -->
		<param name="initial_percent" value="100" />
		<!-- Max current the battery can provide (milliamps hour), default = 7000 -->
		<param name="discharge_current" value="7000" />
		<!-- Max current the charger can provide (milliamps hour), default = 2400 -->
		<param name="recharge_current" value="520000" />
		<!-- Current required by the robot base without the motors (milliamps hour), default = 200 -->
		<param name="base_power_consumption" value="420000" />
		<!-- Current required by the robot motors (milliamps hour), default = 5000 -->
		<param name="motors_power_consumption" value="12600000" />
		<!-- Number of batteries of the robot (connected in parallel), default = 1 -->
		<param name="num_batteries" value="1" />
		<!-- The velocity topic of the robot (geometry_msgs/Twist), default = cmd_vel -->
		<param name="cmd_vel_topic" value="/cmd_vel" />
		<!-- Prints battery info at every second, default = false -->
		<param name="verbose" value="true" />
	</node>
	<!-- Requires rqt_multiplot package: https://github.com/ANYbotics/rqt_multiplot_plugin -->
	<node name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot" args="--force-discover --multiplot-config $(find battery_simulator)/config/rqt_multiplot.xml --multiplot-run-all" />
</launch>
